app/controllers/UserController.scala

```diff
@@ -26,6 +26,12 @@ class UserController @Inject()(userService: UserService, auth: Authentication) e
     userService.getAll.map ( users => Ok(Json.toJson(users)) )
   }
 
+  def searchUsers:Action[Seq[TechFilter]] = auth.UserAction(AccessLevel.Management).async(BodyParsers.parse.json[Seq[TechFilter]]) { r =>
```

Comment:
I think we can extract the parsing of the json in a separate step in the for comprehension. This way we have more control on what error to return to the user in case smth is wrong with the received json